boards/posix/native_posix/cpu_wait.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include "timer_model.h"
+#include <arch/posix/posix_soc_if.h>
+#include <posix_board_if.h>
+#include <posix_soc.h>
+
+#if defined(CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT)
```

Comment:
Only because of the POSIX ARCH selecting it.
The code still compiles without it (note that this function has only been moved (and a comment fixed a bit))

Reply:
Is your comment that because the POSIX arch selects this option, there is no point of having this ifdef (to allow that option from being de-selected later)?

Reply:
My understanding is that the POSIX arch necessarily needs a custom busy wait so deselecting `CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT` would not make a lot of sense. But I am okay with leaving the guard in if you think there is a possible future where the POSIX arch does not need the custom busy wait :-)

Reply:
It quite needs it yes. I can change it to #error if not selected

Reply:
done